File: kennedy-langlotz/src/AST/Visitor.java
package AST;

import AST.KLQNodes.*;
import AST.KLQNodes.ASTString;

/**
 * Created by juriaan on 10-2-15.
 */
public interface Visitor {
    public void visit(Questionnaire node);
    public void visit(Question node);
    public void visit(ASTString node);
}

File: carlos.cirello/qlang/interpreter/ast/visitor.go
package ast

// Visitor is the base visitor struct which all other Executer AST tree are
// injected for traversal
type Visitor struct {
	Tree Executer
}

// NewVisitor factory for Visitor AST traversal struct
func NewVisitor(tree Executer) *Visitor {
	return &Visitor{
		Tree: tree,
	}
}

// Visit type switch through all possible root AST node types
func (v *Visitor) Visit(node Acceptable) {
	node.Accept(v)
}
